src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/DiagnosticExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Testing.Extensions
+{
+    internal static class DiagnosticExtensions
+    {
+        private static readonly MethodInfo? IsSuppressedGetMethod = typeof(Diagnostic).GetProperty("IsSuppressed")?.GetMethod;
+
+        public static bool TryGetIsSuppressed(this Diagnostic diagnostic, out bool value)
```

Comment:
💭 It probably isn't necessary to make this `TryGetIsSuppressed`. It could instead follow the pattern from `ProjectExtensions.AnalyzerConfigDocuments`:

1. Cache a `Func<Diagnostic, bool>` instead of `MethodInfo`
2. If the `IsSuppressed` property isn't present, treat its value as `false` instead of missing (diagnostics could not be suppressed prior to diagnostic suppressors)